heidrun/nypl_mods.rb

```diff
@@ -0,0 +1,285 @@
+                                    .match_attribute(:type, 'text')
+                                    .map { |v| v.value.downcase }
+                                    .none? { |role_term|
+                                      NON_CONTRIBUTOR_ROLES.include?(role_term)
+                                    }
+                        },
+                as: :contrib do
+      providedLabel contrib.field('namePart')
+    end
+
+    # <name ...> <namePart>[value]</namePart> (When roleTerm is any of these:
+    # Architect; Artist; Author; Cartographer; Composer; Creator; Designer;
+    # Director; Engraver; Interviewer; Landscape architect; Lithographer;
+    # Lyricist; Musical director; Performer; Project director; Singer;
+    # Storyteller; Surveyor; Technical director; Woodcutter) See entire list
+    # here: http://www.loc.gov/marc/relators/relaterm.html
+    creator class: DPLA::MAP::Agent,
+            each: record.field('name')
+                    .select { |name|
+                      name['role']
+                                .field('roleTerm')
+                                .match_attribute(:type, 'text')
+                                .map { |v| v.value.downcase }
+                                .all? { |role_term|
+                                  CREATOR_ROLES.include?(role_term)
+                                }
+                    },
+            as: :contrib do
+      providedLabel contrib.field('namePart')
+    end
+
+    date record.map(&build_dates).flatten
+
+    description record.fields(['note'], %w(physicalDescription note))
+
+    extent record.field('physicalDescription', 'extent')
+
+    dcformat record.field('physicalDescription', 'form')
+
+    genre record.field('genre').select { |genre|
+      genre.value.downcase =~ /book|periodical|magazine/
+    }
+
+    # Top-level <identifier> when displayLabel OR type attribute contain the
+    #   following values: local_imageid isbn isrc isan ismn iswc issn uri urn
+    identifier record.field('identifier')
+                .select { |val|
+                  pattern = Regexp.union('local_imageid', 'isbn', 'isrc',
+                                         'isan', 'ismn', 'iswc', 'issn',
+                                         'uri', 'urn')
+                  val.node['displayLabel'] =~ pattern ||
+                    val.node['type'] =~ pattern
+                }
+
+    language class: DPLA::MAP::Controlled::Language,
+             each: record.field('language', 'languageTerm')
+                     .match_attribute(:type, 'code')
+                     .match_attribute(:authority, 'iso639-2b'),
+             as: :lang do
+      providedLabel lang
+    end
+
+    spatial class: DPLA::MAP::Place,
+            each: record.field('place', 'placeTerm'),
+            as: :place do
+      providedLabel place.field('geographic')
+    end
+
+    # TODO: Might need a deduping enrichment here, as providedLabel might be
```

Comment:
Does [DedupNodes](https://github.com/dpla/KriKri/blob/develop/lib/krikri/enrichments/dedup_nodes.rb#L2-L31) fit the bill?